grc/gui_qt/components/window.py

```diff
@@ -66,7 +66,7 @@
 
 
 class MainWindow(QtWidgets.QMainWindow, base.Component):
-    def __init__(self):
+    def __init__(self, file_path=[]):
```

Comment:
Setting a default `= []` argument *can* be tricky in situations where the function modifies the parameter; the default parameter is only constructed once and then it stays the same object forever. For example:

```python
def print_list(printme = []):
  printme += ["end"]
  print(printme)

print("first call:")
print_list()

print("second call:")
print_list()
```

would print
```text
first call:
['end']
second call:
['end', 'end']
```

Reply:
The typical workaround is to use a `None` default parameter:
```python
def better_print_list(printme = None):
  if not printme:
   printme = []
  printme += ["end"]
  print(printme)
```

Reply:
Thanks for your clarification. I used [] as default, as this is just the value you get from the cli if you enter no grc file.  So I can just add file_path to files_open. And as you mentioned file_path will not be changed in __init__